ldms/src/store/kokkos_appmon/kokkos_appmon.c

```diff
@@ -424,16 +424,23 @@ static int constructor(ldmsd_plug_handle_t handle)
 
 static void destructor(ldmsd_plug_handle_t handle)
 {
-	if (sos)
-		sos_container_close(sos, SOS_COMMIT_ASYNC);
-	if (root_path)
-		free(root_path);
+	ovis_log(mylog, OVIS_LDEBUG, "term %s\n", ldmsd_plug_cfg_name_get(handle));
+        if (sos)
+                sos_container_close(sos, SOS_COMMIT_ASYNC);
+
+        free(root_path);
+        root_path = NULL;
+        free(stream);
+        stream = NULL;
         free(plugin_config_name);
 }
 
-struct ldmsd_plugin ldmsd_plugin_interface = {
-	.config = config,
-	.usage = usage,
-        .constructor = constructor,
-        .destructor = destructor,
+struct ldmsd_store ldmsd_plugin_interface = {
+	.base = {
+		.type = LDMSD_PLUGIN_STORE,
+		.config = config,
+		.usage = usage,
+		.constructor = constructor,
+		.destructor = destructor,
+	},
```

Comment:
@spwalto do none of the stores publish a store() callback function?

Reply:
@tom95858 these are all really just subscribers that store to sos from specific schema stuff. so there's no callback for a metric set handling.

Reply:
@baallan, ok, that's what I figured. They are driven of a receive callback, not a metric set update.